src/runtime/CollectionWrappers/IterableWrapper.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Collections.Generic;
+using System.Collections;
+
+namespace Python.Runtime.CollectionWrappers
+{
+    internal class IterableWrapper<T> : IEnumerable<T>
+    {
+        protected PyObject iterObject;
+        protected PyObject pyObject;
+
+        public IterableWrapper(PyObject pyObj)
+        {
+            pyObject = pyObj;
```

Comment:
I recommend check the argument for `null`

Reply:
Why is this resolved? Is there a legitimate scenario for `IterableWrapper` wrapping `None`/`null`?

Reply:
@lostmsu This is how I interpreted your comment.  did you mean to say I should throw on a null input?

Reply:
@koubaa this is right. Though you should throw `ArgumentNullException`